### PR TITLE
TEST-#6477: Update ASV to 0.5.1

### LINKED ISSUE #6477: Update ASV to 0.5.1

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6477 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: asv_bench/asv.conf.dask.json [hunk 1]
```diff
@@ -13,31 +13,9 @@
     // project being benchmarked
     "repo": "..",
 
-    // The Python project's subdirectory in your repo.  If missing or
```
Comment (anmyachev) [inline asv_bench/asv.conf.dask.json:16]: Due to the simplicity of the changes, I also want to remove unnecessary comments in the configs, in **this** pull request.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -234,11 +234,7 @@ jobs:
           miniforge-version: latest
           use-mamba: true
       - name: ASV installation
-        run: |
-          # FIXME: use the tag or release version of ASV as soon as it appears;
-          # The ability to build a conda environment by specifying yml file has not
-          # yet appeared in the release versions;
-          pip install git+https://github.com/airspeed-velocity/asv.git@ef016e233cb9a0b19d517135104f49e0a3c380e9
+        run: pip install asv==0.6.0
```
Comment (vnlitvinov) [inline .github/workflows/ci.yml:237]: please sync the changes and PR/issue title - this line installs `asv 0.6.0` but title claims `0.5.1` 🙃

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -253,27 +249,21 @@ jobs:
               asv machine --yes
 
               # check Modin on Ray
-              asv run --quick --strict --show-stderr --launch-method=spawn \
+              asv run --quick --show-stderr --launch-method=spawn \
```
Comment (vnlitvinov) [inline .github/workflows/ci.yml:252]: what did `--strict` do and why do you want to remove it?

**Review by vnlitvinov — COMMENTED** [review]
@anmyachev CI is failing now on ASV benchmarks, something is not installing correctly...

File: .github/workflows/ci.yml (hunk 2 above)
Comment (anmyachev) [reply .github/workflows/ci.yml]: I just wanted to try the mamba plugin, it doesn't work yet, we have to wait for 0.6.1. Reverted changes to 0.5.1.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (anmyachev) [reply .github/workflows/ci.yml]: reverted

File: environment-dev.yml [hunk 4]
```diff
@@ -62,12 +62,12 @@ dependencies:
   - pandas-stubs>=2.0.0
 
   - pip:
+      - asv==0.5.1
```
Comment (anmyachev) [inline environment-dev.yml:65]: Let's leave the installation with pip for now, I vaguely remember some problems when installing via conda.

**Review by vnlitvinov — APPROVED** [review]
LGTM!

**Pull request merged**